### PR TITLE
 NMS-9899: Add a new WsMan WQL-based detector

### PR DESCRIPTION
**Pull request by dino2gnt**
* Jira: https://issues.opennms.org/browse/NMS-9899

Add a new WsMan detector that can use WQL filters against an
endpoint to detect arbitrary services.

### CODE REVIEW

**Comment by j-white** [comment]
Looks great. The only issue I see is that the factory is not exposed via the Blueprint, so the detector won't be available on Minion.

Try adding an entry similar to existing one for the `WsManDetectorFactory` in: `opennms/features/wsman/src/main/resources/OSGI-INF/blueprint/blueprint.xml`

**Comment by dino2gnt** [comment]
As requested

**Comment by j-white** [comment]
Pushed to `pr/jira/NMS-9899` for a Bamboo run.

**Comment by j-white** [comment]
Build passed.

@dino2gnt can you also add an entry for this detector to: https://github.com/OpenNMS/opennms/blob/develop/smoke-test/src/test/java/org/opennms/smoketest/minion/DetectorsCommandIT.java#L117

This will test to ensure that the detector loads on Minion successfully.

**Comment by dino2gnt** [comment]
Sufficient?

**Comment by indigo423** [comment]
@dino2gnt thank you very much 🤘

**Comment by j-white** [comment]
I updated `DetectorsCommandIT` to expect the name to be `WsManWQLService` as defined in the detector instead of `WsManWQL`, so that the test would pass.

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/wsmanwql.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+===== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.wsman.WsManWQLDetector`
+|===
+
+===== Configuration and Usage
+
+.Parameters for the <DETECTOR-NAME-HERE>
```
Comment (pioto) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/wsmanwql.adoc]: "<DETECTOR-NAME-HERE>"?

Comment (dino2gnt) [reply opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/wsmanwql.adoc]: Same as opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/wsman.adoc, which was the template for the wql detector doc.  if it's wrong, it's wrong there too :)

**Pull request merged**